inc/class-settings.php

```diff
@@ -82,44 +90,81 @@ function cwp_frequency_render() {
 
 		$options = get_option( 'cwp_settings' );
 		?>
-		<select name='cwp_settings[cwp_frequency]'>
-			<option value='1' <?php selected( $options['cwp_frequency'], 1 ); ?>>Daily</option>
-			<option value='2' <?php selected( $options['cwp_frequency'], 2 ); ?>>Weekly</option>
-			<option value='3' <?php selected( $options['cwp_frequency'], 3 ); ?>>Monthly</option>
-		</select>
+        <select name='cwp_settings[cwp_frequency]'>
+            <option value='1' <?php selected( $options['cwp_frequency'], 1 ); ?>>Daily</option>
+            <option value='2' <?php selected( $options['cwp_frequency'], 2 ); ?>>Weekly</option>
+            <option value='3' <?php selected( $options['cwp_frequency'], 3 ); ?>>Monthly</option>
+        </select>
 
 		<?php
 
 	}
 
+	function cwp_optin_render() {
+
+		$options = get_option( 'cwp_settings' );
+		?>
+        <input type="text" name="cwp_settings[cwp_notify]" value="<?php echo $options['cwp_notify']; ?>">
+		<?php
+
+	}
 
 	function cwp_template_render() {
 
 		$options = get_option( 'cwp_settings' );
 		?>
-		<textarea cols='60' rows='15'
-		          name='cwp_settings[cwp_template]'><?php echo $options['cwp_template']; ?></textarea>
+        <textarea cols='60' rows='15'
+                  name='cwp_settings[cwp_template]'><?php echo $options['cwp_template']; ?></textarea>
 		<?php
 
 	}
 
 	function cwp_options_page() {
 
 		?>
-		<form action='options.php' method='post'>
+        <form action='options.php' method='post'>
 
-			<h2>Custom WP Notify</h2>
+            <h2>Custom WP Notify</h2>
 
 			<?php
 			settings_fields( 'cwpOptions' );
 			do_settings_sections( 'cwpOptions' );
 			submit_button();
 			?>
 
-		</form>
+        </form>
 		<?php
 
 	}
 
+	// Create the [cpw_notify] shortcode
+	function cwp_shortcode( $atts ) {
```

Comment:
what is the reason to have this function part of the Settings Class?

Reply:
Hadn't given it much thought as to where it should live. This one and the one below ( cwpOptIn function). Where would you prefer to see those?

Reply:
`class-shortcodes.php`